streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1836,6 +1842,36 @@ public void maybeThrowTaskExceptionsFromProcessingThreads() {
         }
     }
 
+    private boolean transactionBuffersExceedCapacity = false;
+
+    boolean transactionBuffersExceedCapacity() {
+        return transactionBuffersExceedCapacity;
+    }
+
+    boolean transactionBuffersWillExceedCapacity() {
+        final boolean transactionBuffersAreUnbounded = maxUncommittedStateBytes < 0;
```

Comment:
nit: avoid implicit assumptions about the value of sentinel constants. For example what if we need to introduce another sentinel value that means something other than unbounded, and give it a value of -2? Best to define a static constant like `UNBOUNDED_UNCOMMITTED_BYTES = -1` and then check like this:
```suggestion
        final boolean transactionBuffersAreUnbounded = maxUncommittedStateBytes == UNBOUNDED_UNCOMMITTED_BYTES;
```

Reply:
And if you define the UNBOUNDED_UNCOMMITTED_BYTES constant in StreamsConfig, perhaps next to the config definition itself, then users can easily discover and use the variable for clear code, rather than hardcoding "-1" and then coming back a year later and wondering what that value means